influxdb/client.go

```diff
@@ -243,6 +244,12 @@ func (c *HTTPClient) Update(new Config) error {
 		if tr == nil {
 			tr = old.Transport
 		}
+		tr.DialContext = (&net.Dialer{
+			Timeout:   new.Timeout, // I am not sure if this is the right value to set it to
+			KeepAlive: new.Timeout, // I am not sure if this is the right value to set it to
```

Comment:
I'd tend to think that the `KeepAlive` should be set the same here as it is elsewhere, and not necessarily be the same as the timeout. Since it's being set to 30 sec everywhere else to match `http.DefaultClient`, it would make sense to me to have it set to that here as well.

Question: Is it possible to call this function with a new config that has a `nil` or zero-value duration for the timeout? I'm thinking that would make the timeout be unlimited for the dialer (and the client), and am wondering if that's desirable.